### DISCUSSION TITLE
Best way to configure etcd metrics port.

### DISCUSSION THREAD

**Discussion by evannorstrand1** [rancher/rke2#4024] [Q&A]
I have taken over managing new hardware that I am going to be provisioning with RKE2. 

Port 2381 is in use by HP monitoring software that I can not remove. Etcd is having it's metrics listened to on port 2381.

"listen-metrics-urls: http://127.0.0.1:2381"

How do I change the port to 2382 or something similar?

**Comment by brandond** [comment]
https://docs.rke2.io/reference/server_config?_highlight=etcd-arg

**Reply by evannorstrand1** [reply]
I attempted this @brandond and it didn’t take effect. The etcd config didn’t change nor did the pod manifest

**Reply by evannorstrand1** [reply]
@brandond - following up on this, i've tried multiple different combinations to get this value changed. Do you have any working examples?

**Reply by brandond** [reply]
How are you passing the flag? Can you provide an example of any of the combinations you've tried? It seems to work for me; when I start with `--etcd-arg=listen-metrics-urls=http://127.0.0.1:2382` or the equivalent config file option, I see:

```console
root@rke2-server-1:/# grep listen-metrics /var/lib/rancher/rke2/server/db/etcd/config
listen-metrics-urls: http://127.0.0.1:2382
```

**Reply by mehdi-aghayari** [reply]
Hi,
in /etc/rancher/rke2/config.yaml file can below configure:
```
etcd-arg:
  - "listen-metrics-urls=http://0.0.0.0:2381"
```
  
  if you need call etcd target by prometheus set 0.0.0.0 for IP.
